src/main/java/com/vaadin/starter/skeleton/spring/GreetService.java

```diff
@@ -0,0 +1,14 @@
+package com.vaadin.starter.skeleton.spring;
+
+import java.io.Serializable;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class GreetService implements Serializable {
+
+    public String greet(String name) {
+        return "Hello " + (name != null && !"".equals(name)? name : "anonymous user");
```

Comment:
`!"".equals(name)` should be `!name.isEmpty()`. 

Overall this would be much more readable as
```
if (name == null || name.isEmpty()) {
  return "Hello anonymous user";
} else { 
  return "Hello "+ name;
}
```